components/Main/NewsletterSignUp/SignUpForm/SignUpForm.module.css

```diff
@@ -0,0 +1,18 @@
+.form > * {
+  border: 1px solid gray;
+  font-size: 0.9rem;
+  line-height: var(--spaceOneRatioBigger);
+  margin: 0;
+  padding: calc(var(--spaceThreeRatiosSmaller) - 1px) calc(var(--spaceTwoRatiosSmaller) - 1px);
```

Comment:
Very specific styling here (to the pixel...) what is the intent?

Reply:
Oh, I replace the default focus outline with a wider(2px) border:
```css
.input:focus {
  border: 2px solid black;
  outline: none;
  padding: calc(var(--spaceThreeRatiosSmaller) - 2px) calc(var(--spaceTwoRatiosSmaller) - 2px);
}
```
So normally, an input element only has a 1px border, when focus, it has a 2ps border. Then the padding needs to shrink, so the input element won't grow bigger.

Reply:
And I also set that padding on the button element, so it is at the same height as the input element.

Reply:
Ok how about something like this (which effectively achieves that result):
```
.button,
.input {
  border: 1px solid gray;
  font-size: 0.9rem;
  line-height: var(--spaceOneRatioBigger);
  padding: calc(var(--spaceThreeRatiosSmaller)) calc(var(--spaceTwoRatiosSmaller));
}

.button:hover {
  background-color: black;
  color: white;
}

.input:focus {
  border: 1px solid black;
  outline: none;
  box-shadow: inset 0 0 0 1px black;
}
```

Reply:
Using an inset `box-shadow` we can get the "2px" border. One thing to be aware of here is that we are setting the focus style on the input but not the button which still has the user agent focus ring.

Reply:
The other thing to consider is that when setting typography styles like the `font-size` and `line-height` we want to consider whether these should be global or whether this component does require special typography styles.

Reply:
If we want (and likely do) all buttons and inputs to at least start with the same base styles we should probably create these as components with the respective styles such as the typography, border, focus, and hover styles set in those.

Reply:
Box-shadow is cool! Decoupled with the padding. Thanks for this trick. 
As to the generic section, button and input component, I will create them in the other PR and utilize them on this NewsLetter component after they merged.

Reply:
I created a ticket on the Trello board.